edweiss-app/app/(app)/calendar/index.tsx

```diff
@@ -6,85 +6,119 @@ import { useAuth } from '@/contexts/auth';
-      const assignmentsData: { [key: string]: any[] } = {};
-      const todoData: { [key: string]: any[] } = {};
+      if ((courses as unknown as CourseWithId[]).length === 0) return;
+
+      try {
+        const assignmentsData: { [key: string]: any[] } = {};
+        const todoData: { [key: string]: any[] } = {};
+
+        await Promise.all(
+          courses.map(async (course) => {
+            const assignmentsRef = CollectionOf(`courses/${(courses as unknown as CourseWithId).id}/assignments`);
+            const todoRef = CollectionOf(`users/${authUserId}/todos`);
+
+            const [assignmentsSnapshot, todoSnapshot] = await Promise.all([
+              assignmentsRef.get(),
+              todoRef.get(),
+            ]);
+
+            assignmentsData[(courses as unknown as CourseWithId).id] = assignmentsSnapshot.docs.map((doc) => doc.data());
+            todoData[(courses as unknown as CourseWithId).id] = todoSnapshot.docs.map((doc) => doc.data());
+          })
+        );
+
+        setAssignmentsByCourse(assignmentsData);
+        setTodoByCourse(todoData);
+        setLoading(false);
+      } catch (error) {
+        console.error('Erreur lors du chargement des données Firebase:', error);
+      }
+    };
 
-      for (const course of courses) {
-        const assignmentsRef = CollectionOf(`courses/${course.id}/assignments`);
-        const todoRef = CollectionOf(`users/${auth.authUser?.uid ?? 'default-uid'}/todos`);
+    fetchData();
+  }, [authUserId, courses]);
 
-        const [assignmentsSnapshot, todoSnapshot] = await Promise.all([
-          assignmentsRef.get(),
-          todoRef.get(),
-        ]);
+  return { loading, assignmentsByCourse, todoByCourse };
+};
 
-        assignmentsData[course.id] = assignmentsSnapshot.docs.map(doc => doc.data());
-        todoData[course.id] = todoSnapshot.docs.map(doc => doc.data());
-      }
+const InfinitePaginatedCounterScreen = () => {
+  const auth = useAuth();
+  const dimensions = useWindowDimensions();
+  const { width, height } = dimensions;
 
-      setAssignmentsByCourse(assignmentsData);
-      setTodoByCourse(todoData);
-      setLoading(false);
-    };
+  const { courses, filteredCourses } = useCourseData(auth.authUser?.uid ?? 'default-uid');
+  const { loading, assignmentsByCourse, todoByCourse } = useAssignmentsAndTodos(
+    auth.authUser?.uid ?? 'default-uid',
+    courses
+  );
 
-    if (courses.length > 0) {
-      fetchData();
-    }
-  }, [courses]);
+  const [pages, setPages] = useState(Array.from({ length: 10 }, (_, index) => index - 3));
+  const scrollRef = useRef<FlatList>(null);
 
-  const myCourseIds = myCourses.map(course => course.id);
-  const filteredCourses = courses.filter(course => myCourseIds.includes(course.id));
+  const todos = filteredCourses.flatMap((course) =>
+    (todoByCourse[course.id] || []).map((todo) => ({ id: course.id, data: todo }))
+  );
+
+  const assignments = filteredCourses.flatMap((course) =>
```

Comment:
The variable name is "assignments" event though you ultimately return a list of {id: string, data: Assignment}. Perhaps consider renaming it!

Reply:
(same goes for the todos)